arwen/elrondapi/elrondei.go

```diff
@@ -1390,6 +1402,104 @@ func v1_3_upgradeContract(
 	gasToUse = math.MulUint64(gasSchedule.BaseOperationCost.DataCopyPerByte, uint64(length))
 	metering.UseGas(gasToUse)
 
+	upgradeContract(host, calledSCAddress, code, codeMetadata, value, data, gasLimit)
+}
+
+//export v1_3_upgradeFromSourceContract
+func v1_3_upgradeFromSourceContract(
```

Comment:
This API function must also take the code metadata, just like `upgradeContract` does.